src/doc/trpl/guessing-game.md

```diff
@@ -0,0 +1,991 @@
+
+```rust,ignore
+.read_line(&mut guess)
+```
+
+Here, we call the [`read_line()`][read_line] method on our handle.
+[Method][method]s are like associated functions, but are only available on a
+particular instance of a type, rather than the type itself. We’re also passing
+one argument to `read_line()`: `&mut guess`.
+
+[read_line]: ../std/io/struct.Stdin.html#method.read_line
+[method]: methods.html
+
+Remember how we bound `guess` above? We said it was mutable. However,
+`read_line` doesn’t take a `String` as an argument: it takes a `&mut String`.
+Rust has a feature called ‘[references][references]’, which allows you to have
+multiple references to one piece of data, which can reduce copying. References
+are a complex feature, as one of Rust’s major selling points is how safe and
+easy it is to use references. We don’t need to know a lot of those details to
+finish our program right now, though. For now, all we need to know is that
+like `let` bindings, references are immutable by default. Hence, we need to
+write `&mut guess`, rather than `&guess`.
+
+Why does `read_line()` take a mutable reference to a string? Its job is
+to take what the user types into standard input, and place that into a
+string. So it takes that string as an argument, and in order to add
+the input, it needs to be mutable.
+
+[references]: references-and-borrowing.html
+
+But we’re not quite done with this line of code, though. While it’s
+a single line of text, it’s only the first part of the single logical line of
+code:
+
+```rust,ignore
+        .ok()
+        .expect("Failed to read line");
+```
+
+When you call a method with the `.foo()` syntax, you may introduce a newline
+and other whitespace. This helps you split up long lines. We _could_ have
+done:
+
+```rust,ignore
+    io::stdin().read_line(&mut guess).ok().expect("failed to read line");
+```
+
+But that gets hard to read. So we’ve split it up, three lines for three
+method calls. We already talked about `read_line()`, but what about `ok()`
+and `expect()`? Well, we already mentioned that `read_line()` puts what
+the user types into the `&mut String` we pass it. But it also returns
+a value: in this case, an [`io::Result`][ioresult]. Rust has a number of
+types named `Result` in its standard library: a generic [`Result`][result],
+and then specific versions for sub-libraries, like `io::Result`.
+
+[ioresult]: ../std/io/type.Result.html
+[result]: ../std/result/enum.Result.html
+
+The purpose of these `Result` types is to encode error handling information.
+Values of the `Result` type, like any type, have methods defined on them. In
+this case, `io::Result` has an `ok()` method, which says ‘we want to assume
+this value is a successful one. If not, just throw away the error
+information’. Why throw it away? Well, for a basic program, we just want to
+print a generic error, as basically any issue means we can’t continue. The
+`ok()` method returns a value which has another method defined on it:
```

Comment:
Could this (and the below description of expect) link to the method documentation?